doc/mgr/diskprediction.rst

```diff
@@ -0,0 +1,113 @@
+=====================
+DiskPrediction plugin
+=====================
+
+Disk Prediction plugin is used to collect disk information from Ceph OSD and
+send DiskPrediction server. The disk information includes the following.
+
+- Ceph status
+- I/O operations
+- I/O bandwidth
+- OSD status
+- OSD physical disk smart data
+- Storage utilization
+
+Enabling
+========
+
+Run the following command to enable *diskprediction* module in the Ceph
+environment:
+
+::
+
+    ceph mgr module enable diskprediction
+
+Connection settings
+-------------------
+
+The connection settings can be configured on any machine with the proper cephx
+credentials; they are usually the monitor node with client.admin keyring.
+Run the following command to set up connection between Ceph system and
+DiskPrediction server.
+
+::
+
+    ceph diskprediction config-set <diskprediction_server> <diskprediction_user> <diskprediction_password>
+	
+
+The ``<diskprediction_server>`` parameter is DiskPrediction server name, and it
+could be an IP address if required.
+
+The ``<diskprediction_user>`` and ``<diskprediction_password>`` parameters are the user
+id and password logging in to DiskPrediction server.
+
+
+
+The connection settings can be shown using the following command:
+
+::
+
+    ceph diskprediction config-show
+
+
+Receiving predicted health status from Ceph OSD disk drive
+==========================================================
+
+You can receive predicted health status from Ceph OSD disk drive by using the
+following command.
+
+::
+
+    ceph diskprediction get-predicted-status <osd id>
```

Comment:
What is the purpose of this command?  I think 'ceph device info <devid>' will return the same information (if a life expectancy value is stored)

Reply:
Agree, but I need a command to verify the dump prediction result of the plugin.

Reply:
I think this command, and the underlying data model, should be based on devices, not osd ids.  each device has a unique id in the system, and is consumed by one or more daemons (today they are always osds, but in the very near future it will probably include monitors).  The *device* is the thing that may or may not fail.

The OSD (or other daemon) is only relevant once you start looking at the impact on the system.  So Ceph might prioritize recovery based on how many and which OSDs are affected by looming device failures, which data is stored on them, etc.  For example, a common scenario is an SSD shared by many OSDs.  Once that SSD is suspect we'll want to migrate that data quickly.